io_scene_nif/modules/nif_import/scene/__init__.py

```diff
@@ -38,10 +38,37 @@
 # ***** END LICENSE BLOCK *****
 
 import bpy
-
+from pyffi.formats.nif import NifFormat
+from io_scene_nif.properties.scene import _game_to_enum
 
 def import_version_info(data):
     scene = bpy.context.scene.niftools_scene
     scene.nif_version = data._version_value_._value
     scene.user_version = data._user_version_value_._value
     scene.user_version_2 = data._user_version_2_value_._value
+    
+    #filter possible games by nif version
+    possible_games = set()
+    for game, versions in NifFormat.games.items():
+        if game != '?':
+            if scene.nif_version in versions:
+                possible_games.add(_game_to_enum(game))
+    #filter possible games by user version
+    for game in list(possible_games):
+        if game in scene.USER_VERSION:
```

Comment:
We should do all lookups in one loop, it should work.

Reply:
As it is now, yes. However, I have one question before doing that: to your (or someone else's knowledge), are there games where the user versions can be multiple things? If that is the case, I'd keep it as-is and have checks in between the currently separate loops for len = 1, and set the game early when that is found.
Because right now it's assuming that they're always 0, and discarding if it isn't. I'm not sure that is the case., hence I'd like some input from people who know about more than just Skyrim meshes (me).

Reply:
I think earlier games tend to share Version numbers. Zt2 and oblivion, iirc. They don't have User versions.

Reply:
Not sure what you mean by multiple things?
The newer versions of nif.xml have the breakdown of game mappings done better and even support a "default" version for each game. https://github.com/niftools/nifxml/blob/develop/nif.xml#L58

Reply:
Ah. That breakdown looks very interesting. I take it the double curly brackets around a game mean that that's the 'default' version for that game?
What I meant was, are there games which have associated nif versions, but also use a wide range of non-zero user version or user version 2 (for some reason)? For example, with nif version 4.0.0.0, you know it is from "Freedom Force". However, maybe the user version or user version 2 for nifs from this game could be anything from 1 to say, 100, maybe as an artefact from how they exported it from their software or to denote who made the nif. With the current code in this pull request, you would still discard it and end up not setting the game because the user version is not 0, even though you 'know' it has to be that game based on the nif version.
Of course, if there are no games where that is the case, the point is moot and it can all be rolled up into one loop like Hendrix suggested.

Reply:
I think this is good enough for now.

There are some games that have nifs with the same semantic version as another game. 
So if you read in a file `10.0.1.0` it could be ZT2, CIV4 or OB.
The nif.xml was updated to better support representing this metadata.
`<version id="V10_0_1_0" num="10.0.1.0">{{Zoo Tycoon 2}}, Civilization IV, Oblivion</version>`
The `{{ }}` semantics is to show the `default` version for a specific game, which is what we should be using really when a user selects a "game".

Another reason to get pyffi up to support the updated nif.xml 

Here is the section of the nif.xml where i took the above from - https://github.com/niftools/nifxml/blob/develop/nif.xml#L42